src/util/getTopologicalOrder.ts

```diff
@@ -0,0 +1,56 @@
+export interface Edge {
+	origin: string;
+	target: string;
+}
+
+// Uses the Khan Algorithm loosely
+export default function getTopologicalOrder(inputGraph: ReadonlySet<Edge>): string[] | Set<Edge> {
+	const graph: Set<Edge> = new Set(inputGraph);
+
+	const order: string[] = [];
+	const vertexesWithNoInEdge: Set<string> = getVertexesWithNoInEdge(graph);
+
+	// These are to avoid issues described in following thread by externalizing deletions (from the loop):
+	// https://github.com/diproche/webinterface/pull/28/files/4647fd1d5b3e9050a532c5c24f98d14643f1661e#r287705437
+	const edgesToRemove: Set<Edge> = new Set();
+	const vertexesToRemove: Set<string> = new Set();
+
+ while (vertexesWithNoInEdge.size !== 0) {
+		vertexesWithNoInEdge.forEach((vertex: string) => {
+			vertexesToRemove.add(vertex);
+			order.push(vertex);
+			graph.forEach((edge: Edge) => {
+					if (edge.origin === vertex) { edgesToRemove.add(edge); }
+					vertexesWithNoInEdge.add(edge.target);
+				});
+			// Externalized Deletion
+			edgesToRemove.forEach((edge: Edge) => graph.delete(edge));
+		});
+		// Externalized Deletion
+		vertexesToRemove.forEach((toRemove: string) => vertexesWithNoInEdge.delete(toRemove));
+		vertexesToRemove.clear();
+	}
+
+	// If there are cycles in the graph the residual graph will be returned (which contains the cycles)
+	// This should be regarded as an error case
+	if (graph.size !== 0 ) { return graph; }
```

Comment:
Sorry, my fault, I wasn't clear enough there.
Please don't *return* the errorneous state, but include it in a thrown error(-message).